demo-feeds/src/adapter.rs

```diff
@@ -50,6 +50,19 @@ impl<'a> Vertex<'a> {
     impl_downcast!(as_feed_content, Content, Self::FeedContent);
 }
 
+impl<'a> Typename for Vertex<'a> {
+    fn typename(&self) -> &'static str {
+        match self {
+            Vertex::Feed(..) => "Feed",
+            Vertex::FeedText(..) => "FeedText",
+            Vertex::ChannelImage(..) => "ChannelImage",
+            Vertex::FeedLink(..) => "FeedLink",
+            Vertex::FeedEntry(..) => "FeedEntry",
+            Vertex::FeedContent(..) => "FeedContent",
+        }
+    }
+}
```

Comment:
As soon as we've stabilized the `trustfall` crate's structure, I'm going to move the examples to an `examples/` directory and then make them take advantage of the new derive macros.

Changes like this one in the demo projects are just to make everything compile for the time being.